sklearn/cluster/mean_shift_.py

```diff
@@ -59,6 +59,9 @@ def estimate_bandwidth(X, quantile=0.3, n_samples=None, random_state=0):
         idx = random_state.permutation(X.shape[0])[:n_samples]
         X = X[idx]
     nbrs = NearestNeighbors(n_neighbors=int(X.shape[0] * quantile))
+    if nbrs.get_params()['n_neighbors'] == 0:
+        raise ValueError("cannot estimate bandwidth, try setting a value "
```

Comment:
The ''quantile'' parameter is not exposed to the user, so I think this message is not particularly helpful. I can add something like "[...] or use more than %d samples" though.